posts/2021/12/conda-recipe-grayskull.md

```diff
@@ -0,0 +1,124 @@
+<!--
+.. title: On Software Packages, Conda and Grayskull
+.. slug: conda-recipe-grayskull
+.. date: 2021-12-08 10:00:00 UTC+00:00
+.. author: Mahe Iram Khan
+.. tags: Conda, conda-forge, recipes, Grayskull, packaging, Python
+.. category:
+.. link:
+.. description:
+.. type: text
+.. previewimage: /images/2021/12/grayskull.jpg
+-->
+
+Python might be the most popular snake out there, but most of us have also heard of that other serpent: Conda. And some of us have wondered what it really is. In this post we’ll learn about Conda, software packages and package recipes. Most importantly we’ll learn about Grayskull — a conda recipe generator.
+
+<!-- TEASER_END -->
+
+Hey! I’m [Mahe](https://twitter.com/IramMahe), a Computer Engineering student from India.
+During the summer of 2021 I was an intern at Quansight Labs and I worked on a project called ‘Grayskull’.
+Before we learn about Grayskull, as promised, I’ll talk about software packages and Conda.
+
+## Software Packages
+A software package is an installable piece of code that somebody wrote and published for others to use. In other words, you can install packages so you can benefit from the code others have written without reinventing the wheel.
+
+```python
+import numpy
+```
+
+That’s you ‘importing’ the package numpy into your code.
+
+## Hosting packages
+Once you have written a package you might want to publish it so that others can download it and use it. Depending on the packaging technology you are using, the online location where these packages are made available will receive a different name. Typically, users will know about hosting packages on [PyPI](https://pypi.org/). In the Conda world, packages are hosted on _channels_.
+Channels are like warehouses of packages.
+
+## Conda
+[Conda](https://conda.io/en/latest/index.html) is an OS-agnostic package and environment manager with great popularity in the Python world and data science adjacent libraries. As such, it will allow you to manage the environments and dependencies of your packages and generate the needed context for your project to run successfully on a variety of machines.
+
+Its companion project, Conda-build, is a set of commands and tools that lets you build your own packages for Conda. To create a package with Conda-build, you need to provide a _recipe_: the packaging metadata and build instructions for that specific package. Minimally, a recipe contains a `meta.yaml` file that describes:
+
+- The package name and version
+- Where to get its source code from
+- Its dependencies
+- How to build it
+- Some other metadata
+
+You can learn more about recipes [here](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#meta-yaml).
+
+<p align="center">
+    <img
+     alt="Cute representation of a package recipe"
+     src="/images/2021/12/recipe_animation.png" />
+    <br />
+    <i>Cute representation of a package recipe</i>
```

Comment:
The alt-text should be a bit more descriptive. For example: "a drawing showing a curled up piece of paper with recipe contents on it. The items listed are the ones mentioned previously."

Reply:
@ForgottenProgramme this is still pending, right?

Reply:
Done.